XMonad/Hooks/DynamicLog.hs

```diff
@@ -746,6 +764,197 @@ xmobarStripTags tags = strip [] where
+--
+-- > basePropSB = statusBarPropConfig "xmobar" myPP -- default key: mod-b
+-- > mySB = (withAction (sendMessage ToggleGaps *> sendMessage ToggleStruts) <$> basePropSB >>=) . makeStatusBar
+-- > main = xmonad =<< mySB myConf
+--
+-- The different statusBar functions rely on the following status bar configs:
+--
+-- * 'statusBarProp' relies on 'statusBarPropConfig'
+--
+-- * 'statusBarPropTo' relies on 'statusBarPropConfig'
+--
+-- * 'statusBar' relies on 'statusBarHandleConfig'
+--
+-- * 'statusBar'' relies on 'statusBarHandleConfig''
+--
+-- The main difference is that the functions returning a status bar config do not require a keybinding
+-- and use @mod-b@. Like the toggle action, this can be customized with 'withKey'.
+
+-- | This datataype abstracts a status bar to provide a common interface functions like 'statusBar'
+-- or 'statusBarProp'. Once defined, a status bar can be incorporated in 'XConfig' by using 'makeStatusBar',
+-- which takes care of all the necessary plumbing.
+data StatusBarConfig = StatusBarConfig { sbLogHook     :: X()
+                                         -- ^ How to log to the status bar.
+                                       , sbStartupHook :: X()
+                                         -- ^ How to start the status bar.
+                                       , sbCleanupHook :: X()
+                                         -- ^ How to kill the status bar when xmonad is restarted.
+                                         -- This is useful when the status bar is not started
+                                         -- with a pipe.
+                                       , keyToggle :: XConfig Layout -> (KeyMask, KeySym)
+                                         -- ^ Which keybinding to make the status bar disappear.
+                                       , toggleAction  :: X()
+                                         -- ^ How to toggle the status bar.
+                                       }
+
+-- | Per default, all the hooks do nothing. The 'keyToggle' is set to @mod-b@ and
+-- the 'toggleAction' to @sendMessage ToggleStruts@
+instance Default StatusBarConfig where
+    def = StatusBarConfig { sbLogHook     = mempty
+                          , sbStartupHook = mempty
+                          , sbCleanupHook = mempty
+                          , keyToggle     = toggleStrutsKey
+                          , toggleAction  = sendMessage ToggleStruts
+                          }
+
+-- | Returns a 'StatusBarConfig' with the given 'toggleAction'. This might
+-- be useful when using a more complex layout in which @sendMessage ToggleStruts@
+-- is not enough to make the status bar disappear.
+-- For example, if you use "XMonad.Layout.Gaps", you might want to also toggle the
+-- gaps:
+--
+-- > mySB = (withAction (sendMessage ToggleGaps *> sendMessage ToggleStruts) <$> statusBarPropConfig "xmobar" myPP >>=)
+-- >     . makeStatusBar
+-- > main = xmonad =<< mySB myConf
```

Comment:
It's kinda neat to have these examples all be single lines, but this goes a bit too far I think.  Especially someone who may not be that confident in their ability to read operator soup may be pretty lost here.  It may be more worthfile to show a nicely structured `mySB`

Reply:
You're right, could you check if it's better now?